source/FreeRTOS_DNS_Parser.c

```diff
@@ -786,15 +786,7 @@
                         {
                             BaseType_t xCallbackResult;
 
-                            #if ( ipconfigUSE_IPv6 != 0 )
-                                {
-                                    xCallbackResult = xDNSDoCallback( pxSet, ( ppxAddressInfo != NULL ) ? *( ppxAddressInfo ) : NULL );
-                                }
-                            #else
-                                {
-                                    xCallbackResult = xDNSDoCallback( pxSet, pxSet->ulIPAddress );
```

Comment:
This is a good find. Thank you.
Please check once why this "integer to pointer" assignment is not being treated as an error in our build.